File: Striver/JAVA/implement_stack_using_arrays.java
class Stack{
    private int[] stack;
    private int top;
    private int size;
    public Stack(int s){
        size = s;
        top = -1;
        stack = new int[size];
    }
    void push(int val){
        if(top == size-1){
            System.out.println("Cannot Push , Stack Overflow!");
            return;
        }
        stack[++top] = val;
    }
    void pop(){
        if(top == -1){
            System.out.println("Cannot Pop, Stack Underflow!");
            return;
        }
        top--;
    }
    int top(){
        if(top == -1){
            System.out.println("Stack Is Empty!");
            return -1;
        }
        return stack[top];
    }
    boolean isEmpty(){
        return top == -1;
    }
}
public class implement_stack_using_arrays {
    public static void main(String[] args) {
        Stack st = new Stack(5);
        st.push(10);
        st.push(20);
        st.push(30);
        System.out.println("Top: " + st.top());
        st.pop();
        System.out.println("Top After Pop : " + st.top()); 
    }
}

File: Striver/CPP/implement_stack_using_arrays.cpp
#include<iostream>
#include<vector>

using namespace std;

class Stack{
    private:
        vector<int> stack;
        int size;
        int top;
    public:
    Stack(int s){
            size = s;
            top = -1;
            stack.resize(size);
    };
    void push(int val){
        if(top == size - 1){
            cout<<"Stack Overflow, Cannot Push!"<<endl;
            return;
        };
        stack[++top] = val;
    };
    void pop(){
        if(top == -1){
            cout<<"Stack Underflow, Cannot PoP!"<<endl;
            return;
        };
        top--;
    };
    int peek(){
        if(top == -1){
            cout<<"Stack is empty!"<<endl;
            return -1;
        };
        return stack[top];
    };
    bool isEmpty(){
        return top == -1;
    };
};

int main(){
    Stack st(5);
    st.push(10);
    st.push(20);
    st.push(30);
    cout << "Top: " << st.peek() << endl;
    st.pop();
    cout << "Top after pop: " << st.peek() << endl;
    return 0;
}